src/components/__tests__/calculator.test.js

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { render, fireEvent, screen } from "@testing-library/react";
+import Calculator from "../Calculator";
+
+test("renders the calculator component", () => {
+  render(<Calculator />);
+  const headingElement = screen.getByText(/Let's do some math!/i);
+  expect(headingElement).toBeInTheDocument();
+});
+
+test("adds two numbers correctly", () => {
+  render(<Calculator />);
+  const button1 = screen.getByText("1");
+  const button2 = screen.getByText("2");
+  const addButton = screen.getByText("+");
+  const equalsButton = screen.getByText("=");
+  const result = screen.getByTestId("result");
+
+  fireEvent.click(button1);
+  fireEvent.click(addButton);
+  fireEvent.click(button2);
+  fireEvent.click(equalsButton);
+
+  expect(result).toHaveTextContent("3");
+});
+
+test("subtracts two numbers correctly", () => {
+  render(<Calculator />);
+  const button4 = screen.getByText("4");
+  const button3 = screen.getByText("3");
+  const subtractButton = screen.getByText("-");
+  const equalsButton = screen.getByText("=");
+  const result = screen.getByTestId("result");
```

Comment:
- [ ] This line is the reason for another failing test. kindly consider fixing it before submitting for review as we should pass all the tests as per project requirements.